csm/cli/cortxcli.py

```diff
@@ -113,6 +117,32 @@ def login(self):
             Log.critical(f"{self.username}:{e}")
             self.do_exit(self.some_error_occured)
 
+    def extract_token(self) -> bool:
+        """
+        Function checks the existence of session token in pam output file and validate it.
+        :return:Boolean
+        """
+        user =  getuser()
+        header_file = const.PAM_RESPONSE_HEADER_FILE + user
+        if not os.path.exists(header_file):
+            return False
+        try:
+            with open(header_file, "r") as pam_response_file:
+                header_data = pam_response_file.read()
+            os.remove(header_file)
```

Comment:
What if you have read the session token and failed to delete the file. As per the current logic you will return false resulting in no login. 
It is better to extract session token and save it in self._session_token . After that perform the delete operation.

Reply:
@pawankumarsrivastava  done.